chain/network/src/routing.rs

```diff
@@ -769,23 +769,23 @@ impl RoutingTable {
         }
         debug!(target: "network", "try_save_edges: We are going to remove {} peers", to_save.len());
 
-        let component_nonce = self.component_nonce;
+        let old_component_nonce = self.component_nonce;
```

Comment:
I think that the original code didn't have the bug, but was very confusing..

If you look at the line 428:
self.component_nonce = 
.map_or(0, |nonce| nonce + 1);

So basically the approach was:
- the "nonce" that is stored in the database, is the last row/nonce that was used to store something there.
Then - on startup (like 428) - we read it from DB and increment.

which means that self.component_nonce is the next nonce that can be used.

Reply:
Ok, then we can close this PR.
Renaming `component_nonce` to `next_unused_component_nonce` should be enough.